fabric-message-api-v1/src/main/java/net/fabricmc/fabric/api/message/v1/ChatDecoratorEvent.java

```diff
@@ -0,0 +1,111 @@
+
+import net.minecraft.network.ChatDecorator;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * A class for registering a {@link ChatDecorator}. Check the chat decorator documentation
+ * for how chat decorators work. Unlike other events, this uses a functional interface that is
+ * provided by the vanilla game.
+ *
+ * <p>This event uses phases to provide better mod compatibilities between mods that add custom
+ * content and styling. Chat decorators with the styling phase will always apply after the ones with
+ * the content phase. When registering the chat decorator, it is recommended to choose one of
+ * the phases from this interface and pass that to the {@link Event#register(Identifier, Object)}
+ * function. If not given, the chat decorator will run in the default phase, which is between
+ * the content phase and the styling phase.
+ *
+ * <p>When implementing a chat decorator, it is <strong>very important that the decorator returns
+ * the same contents between previewing and submission</strong> - otherwise the game discards the
+ * message because it was improperly signed. The most straightforward way of doing this is by
+ * making sure the decorator returns the same result for a given message. Another way to solve this
+ * issue is to cache the decorated message.
+ *
+ * <p>Example of registering a content phase chat decorator:
+ *
+ * <pre><code>
+ * ChatDecoratorEvent.EVENT.register(ChatDecoratorEvent.CONTENT_PHASE, (sender, message) -> {
+ *     // Add smiley face. Has to copy() to get a MutableText with siblings and styles.
+ *     return message.copy().append(" :)");
+ * });
+ * </code></pre>
+ *
+ * <p>Example of registering a styling phase chat decorator:
+ *
+ * <pre><code>
+ * ChatDecoratorEvent.EVENT.register(ChatDecoratorEvent.STYLING_PHASE, (sender, message) -> {
+ *     // Apply orange color to messages sent by server operators
+ *     if (sender != null && sender.server.getPlayerManager().isOperator(sender.getGameProfile())) {
+ *         return CompletableFuture.completedFuture(
+ *             message.copy().styled(style -> style.withColor(0xFFA500)));
+ *     }
+ *     return CompletableFuture.completedFuture(message);
+ * });
+ * </code></pre>
+ */
+public final class ChatDecoratorEvent {
+	private ChatDecoratorEvent() {
+	}
+
+	/**
+	 * The content phase of the event, passed when registering a chat decorator. Use this when
+	 * the chat decorator modifies the text content of the message.
+	 */
+	public static final Identifier CONTENT_PHASE = new Identifier("fabric-message-api-v1", "content");
+	/**
+	 * The styling phase of the event, passed when registering a chat decorator. Use this when
+	 * the chat decorator only modifies the styling of the message with the text intact.
+	 */
+	public static final Identifier STYLING_PHASE = new Identifier("fabric-message-api-v1", "styling");
```

Comment:
IMO we should use just fabric for the namespace in these, uniqueness requirements are low and makes upgrading easier